libnestutil/block_vector.h

```diff
@@ -247,6 +260,9 @@ class BlockVector
    */
   int get_max_block_size() const;
 
+  // TODO: To make BlockVector a complete random access container, it should also implement
+  // max_size(), rbegin(), and rend().
```

Comment:
This should either be done within this PR if straightforward or turned into a follow-up issue, but not stay as a "TODO" comment. If turned into a follow-up issue, would it be an idea to implement these three methods now to just throw an exception?

Reply:
I have created function definitions that trow exceptions and removed the "TODO".